gsl_projet/tasks.py

```diff
@@ -33,9 +33,22 @@ def create_or_update_projet_and_its_simulation_and_programmation_projets_from_do
 
 
 @shared_task
-def create_or_update_projets_and_its_simulation_and_programmation_projets_from_all_dossiers():
-    dossiers = Dossier.objects.exclude(ds_state="")
-    for dossier in dossiers.all():
+def create_or_update_projets_and_its_simulation_and_programmation_projets_from_all_dossiers(
+    batch_size=500,
+):
+    dossiers = Dossier.objects.exclude(ds_state="").values_list("ds_number", flat=True)
+
+    dossier_batches = [
+        dossiers[i : i + batch_size] for i in range(0, len(dossiers), batch_size)
+    ]
+
+    for batch in dossier_batches:
+        create_or_update_projets_batch.delay(list(batch))
+
+
+@shared_task
+def create_or_update_projets_batch(dossier_numbers: list[int]):
```

Comment:
Tu peux utiliser `itertools.batched` (cf [doc](https://docs.python.org/3/library/itertools.html#itertools.batched)) au lieu de bricoler tes batch à la main : 

```suggestion
    from itertools import batched # en début de fichier
    
    for batch in batched(dossiers, batch_size):
        create_or_update_projets_batch.delay(batch) # j'enlève aussi le cast en list


@shared_task
def create_or_update_projets_batch(dossier_numbers: tuple[int]): # je demande un tuple au lieu d'une list
```

Reply:
Ah oui cool ! C'est changé